.github/bin/build-matrix-from-impacted.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+import argparse
+import yaml
+import json
+import logging
+import sys
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Filter test matrix modules using list of impacted modules."
+    )
+    parser.add_argument(
+        "-m",
+        "--matrix",
+        type=argparse.FileType("r"),
+        default=".github/test-matrix.yaml",
+        help="A YAML file with the test matrix",
+    )
+    parser.add_argument(
+        "-i",
+        "--impacted",
+        type=argparse.FileType("r"),
+        default="gib-impacted.log",
+        help="List of impacted modules, one per line",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=argparse.FileType("w"),
+        default=sys.stdout,
+        help="Filename to write JSON output to",
+    )
+    parser.add_argument(
+        "-d",
+        "--debug",
+        action="store_const",
+        dest="loglevel",
+        const=logging.DEBUG,
+        default=logging.WARNING,
+        help="Print lots of debugging statements",
+    )
+
+    args = parser.parse_args()
+    logging.basicConfig(level=args.loglevel)
+    build(args.matrix, args.impacted, args.output)
+
+
+def build(matrix_file, impacted_file, output_file):
+    matrix = yaml.load(matrix_file, Loader=yaml.Loader)
+    impacted = list(filter(None, [line.rstrip() for line in impacted_file.readlines()]))
+    logging.debug("Read matrix: %s", matrix)
+    logging.debug("Read impacted: %s", impacted)
+    include = []
+    for item in matrix.get("include", []):
+        if not any(module in impacted for module in item.get("modules", [])):
+            continue
```

Comment:
Most of the logging right now is about what is included. I'd suggest to log (at info level) the modules/jobs that got excluded instead. It's much easier to reason about whether some job should have been excluded or not. The includes ones anyway will be visible as the running CI jobs.

Reply:
Bump @nineinchnick.

Can we log the excluded ones in addition to the "Include" we log currently.

Reply:
I'm on it.

Reply:
Done, and I've rebased again.